basic-auth/routes/authorRouter.js

```diff
@@ -0,0 +1,97 @@
+const express = require("express");
+const bcrypt = require("bcrypt");
+const User = require("../models/User");
+const router = express.Router();
+const bcryptSalt = 10;
+
+router.get("/signup", (req, res, next) => {
+  res.render("author/signup");
+});
+
+router.post("/signup", (req, res, next) => {
+  const { username, password } = req.body;
+
+  User.findOne({
+    username
+  })
+    .then(user => {
+      console.log(user);
+      if (user !== null) {
+        throw new Error("Username Already exists");
+      }
+
+      const salt = bcrypt.genSaltSync(bcryptSalt);
+      const hashPass = bcrypt.hashSync(password, salt);
+
+      const newUser = new User({
+        username,
+        password: hashPass
+      });
+
+      return newUser.save();
+    })
+    .then(user => {
+      res.render("/");
```

Comment:
Esto os da error. Si queréis renderizar index tendréis que poner `res.render("index")`
si lo que queréis es redireccionar sería `res.redirect("/");` no os confundáis.
Cuando utilizáis render tendréis que especificar el nombre del archivo que queréis renderizar, en cambio si utilizáis redirect tendréis que especificar el destino que quieres que tenga.